### PR TITLE
publish workflow for github actions

### LINKED ISSUE #15: Add GitHub workflow for publishing packages
There should be a GitHub Actions workflow that publishes a package to PyPI whenever create a new tag. This will also required created a PyPI account for the Civic Jabber project and adding the credentials to the list of secrets in GH.

### PR DESCRIPTION
**Pull request by kvnkho**
I know the initial issue said on tag creation but it seems people do it on releases. Do we need it on tags? Releases seem appropriate.

Not quite sure how to test this without publishing to PyPi

### CODE REVIEW

**Comment by MthwRobinson** [comment]
@kvnkho -- Releases work for me. I'll test this locally using the PyPI creds a little later.

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        pip install setuptools wheel twine
+    - name: Test
+      run: make test
+    - name: Build and publish
+      env:
+        TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
+        TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
+      run: |
+        make package
+        twine upload dist/*
```
Comment (MthwRobinson) [inline .github/workflows/publish.yml]: Do you want to add a `make publish` that runs `make package` and then runs `twine upload dist/*` after? Then I can give that a whirl locally with the twine environmental variables set.

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+    - name: Install dependencies
+      run: |
+        make pip-install
+        pip install setuptools wheel twine
```
Comment (MthwRobinson) [inline .github/workflows/publish.yml]: We can probably add these to the dev dependencies. You can add them to `requirements/dev.in` and then run `make pip-compile`.

**Review by MthwRobinson — CHANGES_REQUESTED** [review]
There are a couple of other packaging items that I think we should hit in this PR.

1. `setup.py` doesn't install the dependencies right. We should have it install the  reqs in `requirements/base.txt`
1. Let's change the package name from `data_ingest` to `civic_jabber_ingest`. This command from the root directory of the project should (hopefully) do the trick: `find ./ -type f -exec sed -i -e 's/data_ingest/civic_jabber_ingest/g' {} \;

**Comment by kvnkho** [comment]
i believe this is done.

aside from comments above, i also added init.py files for the setup.py to capture these modules when it calls find_packages().

just a note that i had to use anaconda to install the dependencies failing on windows to test it.

i also feel it's very hard for users if versions are pinned with == rather than >= or <=.

**Comment by MthwRobinson** [comment]
Thanks! One final comment for `setup.py` - I think we also need to add [package data](https://setuptools.readthedocs.io/en/latest/userguide/datafiles.html) to include the config `.yml` files and the Jinja templates in the build. Could you add that in as well?

**Comment by kvnkho** [comment]
Added data files and tested. They are now present in wheel file.

**Review by MthwRobinson — APPROVED** [review]

**Comment by MthwRobinson** [comment]
Working for me locally. LGTM. Thanks!! :-)

**Pull request merged**